esrally/racecontrol.py

```diff
@@ -206,7 +207,10 @@ def setup(self, msg, sender):
                 raise exceptions.SystemSetupError("A distribution version is required. Please specify it with --distribution-version.")
             self.logger.info("Automatically derived distribution version [%s]", distribution_version)
             self.cfg.add(config.Scope.benchmark, "mechanic", "distribution.version", distribution_version)
-
+        
+        name = self.cfg.opts("race", "pipeline")
+        if not (name == "benchmark-only" or track.is_simple_track_mode(self.cfg)):
+            self.track_revision = git.head_revision(track.track_path(self.cfg))
```

Comment:
IMHO invoking `git.head_revision` i.e. an FS+external command operation against a local directory seems out of place for this file.

As we are anyway supporting the `--track-revision` command line argument, we could always populate it for the `cfg` object with the exact value, if unspecified, in a separate module. In [loader.py/GitTrackRepository](https://github.com/elastic/rally/blob/9b9c7e03b79a11414740fa8f543b7a58b162149b/esrally/track/loader.py#L188-L206) and [SimpleTrackRepository](https://github.com/elastic/rally/blob/9b9c7e03b79a11414740fa8f543b7a58b162149b/esrally/track/loader.py#L219) we are doing a bunch of low-level stuff, so this could be a starting point?

Note that we are passing the [cfg object](https://github.com/elastic/rally/blob/9b9c7e03b79a11414740fa8f543b7a58b162149b/esrally/racecontrol.py#L231) gets passed to the actor system so the information should be accessible by actors.